onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -666,22 +666,86 @@ def aten_atanh(self: TFloat) -> TFloat:
     return op.Atanh(self)
 
 
-def aten_atleast_1d(self: TensorType) -> TensorType:
+@torch_op("aten::atleast_1d")
+def aten_atleast_1d(self: Sequence[TTensor]) -> TTensor:
     """atleast_1d(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    @graph()
+    def reshape_to_1d(tensor):
+        shape = op.Shape(tensor)
+        rank = op.Size(shape)
+        if rank == 0:
+            tensor = op.Reshape(tensor, op.Constant(value_ints=[1]))
+        return tensor
+
+    return op.SequenceMap(self, body=reshape_to_1d)
+
+
+@torch_op("aten::atleast_1d")
+def aten_atleast_1d_single_tensor(self: TTensor) -> TTensor:
+    """atleast_1d(Tensor self) -> Tensor"""
+
+    shape = op.Shape(self)
+    rank = op.Size(shape)
+    if rank == 0:
+        self = op.Reshape(self, op.Constant(value_ints=[1]))
+    return self
 
 
-def aten_atleast_2d(self: TensorType) -> TensorType:
+@torch_op("aten::atleast_2d")
+def aten_atleast_2d(self: Sequence[TTensor]) -> TTensor:
     """atleast_2d(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    @graph()
+    def reshape_to_2d(tensor):
+        shape = op.Shape(tensor)
+        rank = op.Size(shape)
+        if rank <= 1:
+            tensor = op.Reshape(tensor, op.Constant(value_ints=[1, -1]))
+        return tensor
+
+    return op.SequenceMap(self, body=reshape_to_2d)
+
+
+@torch_op("aten::atleast_2d")
+def aten_atleast_2d_single_tensor(self: TTensor) -> TTensor:
+    """atleast_2d(Tensor self) -> Tensor"""
+
+    shape = op.Shape(self)
+    rank = op.Size(shape)
+    if rank <= 1:
+        self = op.Reshape(self, op.Constant(value_ints=[1, -1]))
+    return self
 
 
-def aten_atleast_3d(self: TensorType) -> TensorType:
+@torch_op("aten::atleast_3d")
+def aten_atleast_3d(self: Sequence[TTensor]) -> TTensor:
     """atleast_3d(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    @graph()
+    def reshape_to_3d(tensor):
+        shape = op.Shape(tensor)
+        rank = op.Size(shape)
+        if rank <= 1:
+            tensor = op.Reshape(tensor, op.Constant(value_ints=[1, -1, 1]))
```

Comment:
You may want to double-check the downstream usage to make sure whether you want to reshape a tensor of size N to 1x1xN or Nx1x1 ... the above reshapes it to 1xNx1 .... which may be okay, but just wondering.